src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/FhirDataStore.cs

```diff
@@ -211,6 +213,49 @@ public async Task HardDeleteAsync(ResourceKey key, CancellationToken cancellatio
             }
         }
 
+        public async Task<HttpStatusCode> UpsertExportJobAsync(ExportJobRecord jobRecord, CancellationToken cancellationToken = default)
+        {
+            EnsureArg.IsNotNull(jobRecord, nameof(jobRecord));
+
+            try
+            {
+                var result = await _documentClient.UpsertDocumentAsync(
+                _collectionUri,
+                jobRecord,
+                new RequestOptions() { PartitionKey = new PartitionKey(jobRecord.PartitionKey) },
+                disableAutomaticIdGeneration: true,
+                cancellationToken: cancellationToken);
+
+                return result.StatusCode;
+            }
+            catch (DocumentClientException e)
+            {
+                if (e.Error?.Message?.Contains(GetValue(HttpStatusCode.RequestEntityTooLarge), StringComparison.Ordinal) == true)
+                {
+                    throw new ServiceUnavailableException();
```

Comment:
We have `RequestRateExceededException` now. Can we replace this and the one above with the new exception?